lib/jnpr/junos/utils/fs.py

```diff
@@ -268,30 +270,34 @@ def _decode(fs):
     # directory_usage - filesystem directory usage
     # -------------------------------------------------------------------------
 
-    def directory_usage(self, path="."):
+    def directory_usage(self, path=".", depth=0):
         """
         Returns the directory usage, similar to the unix "du" command.
 
-        :returns: approximate directory usage, including subdirectories, in bytes
+        :returns: dict of directory usage, including subdirectories if depth > 0
         """
-        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth="0")
+        BLOCK_SIZE = 512
 
-        used_space = rsp.findtext("directory/used-space")
+        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth=str(depth))
 
-        if used_space is None:
-            return None
+        if rsp.findtext("directory/used-space") is None:
+            # Likely, no such directory
+            raise RpcError(rsp=rsp)
 
-        used_space = used_space.strip()
+        dirs = rsp.xpath("//directory")
+        result = {}
 
-        multiplier = {
-            'B': 1,
-            'K': 1024,
-            'M': 1024**2,
-            'G': 1024**3,
-            'T': 1024**4,
-        }
+        for directory in dirs:
+            dir_name = directory.findtext("directory-name").strip()
+            dir_size = directory.findtext("used-space").strip()
+            dir_blocks = int(directory.find("used-space").get("used-blocks").strip())
+            result[dir_name] = {
+                "size": dir_size,
+                "blocks": dir_blocks,
+                "bytes": dir_blocks * BLOCK_SIZE,
+            }
```

Comment:
We need to account for the possibility that the `<used-space>` element (or `used-blocks` attribute) doesn't exist. I suggest replacing lines 292-298 with:
```
used_space = directory.find('used-space')
if used_space is not None:
    dir_size = used_space.text()
    dir_blocks = used_space.get('used-blocks')
    if dir_blocks is not None:
        dir_bytes = dir_blocks * BLOCK_SIZE
        result[dir_name] = {
            "size": dir_size,
            "blocks": dir_blocks,
            "bytes": dir_bytes,
        }
```